agents/sorenson_dlm/dlm_agent.py

```diff
@@ -0,0 +1,349 @@
+
+
+class DLM:
+    def __init__(self, ip_address, port=9221, timeout=10):
+        self.ip_address = ip_address
+        self.port = port
+        self.comm = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.comm.connect((self.ip_address, self.port))
+        self.comm.settimeout(timeout)
+
+    def send_msg(self, cmd):
+        """
+        Sends a message to the DLM. 'OPC?' causes the DLM to wait for the
+        previous command to complete to being the next
+        """
+        msg = str(cmd) + ';OPC?\r\n'
+        self.comm.send(msg.encode('ASCII'))
+        time.sleep(0.1)
+
+    def rec_msg(self):
+        """
+        Waits for a message from the DLM, typically as a result of a query,
+        and returns it
+        """
+        dataStr = self.comm.recv(1024).decode('ASCII')
+        return dataStr.strip()
+
+    def set_overv_prot(self, voltage):
+        """
+        Sets the overvoltage protection
+        """
+        self.send_msg('*CLS')
+        self.send_msg('*RST')
+        self.send_msg('SOUR:VOLT:PROT {}'.format(voltage))
+
+        self.send_msg('SOUR:VOLT:PROT?')
+        ovp = self.rec_msg()
+        if ovp != str(voltage):
+            print("Error: Over voltage protection not set to requested value")
+            return False
+
+        self.send_msg('STAT:PROT:ENABLE 8')
+        self.send_msg('STAT:PROT:ENABLE?')
+        enb = self.rec_msg()
+        if enb != '8':
+            print('Error: Over voltage protection failed to enable')
+            return False
+
+        self.send_msg('STAT:PROT:EVENT?')
+        event = self.rec_msg()
+        if event != '0':
+            print('Error: Over voltage already tripped')
+            return False
+
+        return True
+
+    def read_voltage(self):
+        """
+        Reads output  voltage
+        """
+        self.send_msg('SOUR:VOLT?')
+        msg = self.rec_msg()
+        return msg
+
+    def read_current(self):
+        """
+        Reads output current
+        """
+        self.send_msg('MEAS:CURR?')
+        msg = self.rec_msg()
+        return msg
+
+    def sys_err_check(self):  # never used
+        """
+        Queries sytem error and returns error byte
+        """
+        self.send_msg('SYST:ERR?')
+        msg = self.rec_msg()
+        return msg
+
+
+class DLMAgent:
+    """Agent to connect to a Sorenson DLM power supply via ethernet.
+
+    Args:
+        ip_address: str
+            IP address of the DLM
+        port: int
+            Port number for DLM; default is 9221
+
+    """
+
+    def __init__(self, agent, ip_address, port=9221, f_sample=2.5):
+        self.agent = agent
+        self.ip_address = ip_address
+        self.port = port
+        self.log = agent.log
+        self.lock = TimeoutLock()
+        self.f_sample = f_sample
+        self.take_data = False
+        self.over_volt = 0.
+        self.initialized = False
+        self._acq_proc_lock = TimeoutLock()
+        self._lock = TimeoutLock()
```

Comment:
There are three different `TimeoutLock`'s here, you only need one. Remove these and use `self.lock` everywhere.